apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -2801,21 +2835,71 @@ def to_anndata(
         )
     measurement = experiment.ms[measurement_name]
 
+    # How to choose index name for AnnData obs and var dataframes:
+    # * If the desired names are passed in, use them.
+    # * Else if the names used at ingest time are available, use them.
+    # * Else use the default/fallback name.
+
+    if obs_id_name is None:
+        # Restore the original index name for outgest. We use JSON for elegant indication of index
+        # name being None (in Python anyway).
+        if _DATAFRAME_ORIGINAL_INDEX_NAME in experiment.obs.metadata:
+            # Maybe 'null' which maps to Pyhton None
+            obs_id_name = json.loads(
+                experiment.obs.metadata[_DATAFRAME_ORIGINAL_INDEX_NAME]
+            )
+        else:
+            obs_id_name = "obs_id"
```

Comment:
to avoid the need for the `in` check, you can `.get` with the default value we want:

```python
if obs_id_name is None:
  obs_id_name = json.loads(experiment.obs.metadata.get(_DATAFRAME_ORIGINAL_INDEX_NAME, '"obs_id"'))
```

(or, to make it the ultimate one-liner, `obs_id_name = obs_id_name or json.loads(...)`)